storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/CombinedKvStoreDao.java

```diff
@@ -417,6 +417,19 @@ public Map<String, Long> getColumnCounts() {
     return columnCounts;
   }
 
+  @Override
+  @MustBeClosed
+  @SuppressWarnings("unchecked")
+  public Stream<UInt64> streamFinalizedStateSlots(final UInt64 startSlot, final UInt64 endSlot) {
+    if (schema.getColumnMap().containsKey("FINALIZED_STATES_BY_SLOT")) {
+      KvStoreColumn<UInt64, BeaconState> column =
+          (KvStoreColumn<UInt64, BeaconState>)
+              schema.getColumnMap().get("FINALIZED_STATES_BY_SLOT");
+      return db.stream(column, startSlot, endSlot).map(ColumnEntry::getKey);
+    }
+    return Stream.empty();
+  }
```

Comment:
I don't think this is right. It should be delegating to `V4FinalizedStateStorageLogic<S> stateStorageLogic` which will know which column to use.